app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyResponse.java

```diff
@@ -13,54 +15,72 @@
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.fasterxml.jackson.databind.node.JsonNodeFactory;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.google.common.collect.Lists;
 
-@DynamoDBTable(tableName = "SurveyResponse")
+@DynamoDBTable(tableName = "SurveyResponse2")
 public class DynamoSurveyResponse implements SurveyResponse {
 
-    private static final String HEALTH_CODE_PROPERTY = "healthCode";
-    private static final String COMPLETED_ON_PROPERTY = "completedOn";
-    private static final String STARTED_ON_PROPERTY = "startedOn";
-    private static final String VERSION_PROPERTY = "version";
     private static final String ANSWERS_PROPERTY = "answers";
     
-    public static final DynamoSurveyResponse fromJson(JsonNode node) {
-        DynamoSurveyResponse survey = new DynamoSurveyResponse();
-        survey.setHealthCode(JsonUtils.asText(node, HEALTH_CODE_PROPERTY));
-        survey.setVersion(JsonUtils.asLong(node, VERSION_PROPERTY));
-        survey.setStartedOn(JsonUtils.asMillisSinceEpoch(node, STARTED_ON_PROPERTY));
-        survey.setCompletedOn(JsonUtils.asMillisSinceEpoch(node, COMPLETED_ON_PROPERTY));
-        survey.setData((ObjectNode)node);
-        return survey;
-    }
-    
-    private String surveyGuid; // stored in dynamo
-    private long surveyCreatedOn; // stored in dynamo
-    private Survey survey; // constructed and returned to the consumer
-    private String identifier;
     private String healthCode;
-    private long startedOn;
-    private long completedOn;
+    private String identifier;
+    
+    // These three are stored in Dynamo.
+    private String surveyKey;
+    private String surveyGuid;
+    private long surveyCreatedOn;
+    
+    // This retrieved and returned as part of the JSON to the consumer
+    private Survey survey;
+    
+    private Long startedOn;
+    private Long completedOn;
     private Long version;
     private List<SurveyAnswer> answers = Lists.newArrayList();
     
-    public DynamoSurveyResponse() {
+    @Override
+    @JsonIgnore
+    @DynamoDBHashKey
+    public String getHealthCode() {
+        return healthCode;
     }
-    
-    public DynamoSurveyResponse(String healthCode, String identifier) {
+    @Override
+    public void setHealthCode(String healthCode) {
         this.healthCode = healthCode;
+    }
+    
+    // Main range key: identifier
+    @Override
+    @DynamoDBRangeKey
+    public String getIdentifier() {
+        return identifier;
+    }
+    @Override
+    public void setIdentifier(String identifier) {
         this.identifier = identifier;
     }
     
+    @JsonIgnore
+    @DynamoDBAttribute
+    @DynamoDBIndexRangeKey(localSecondaryIndexName = "healthCode-surveyKey-index")
+    public String getSurveyKey() {
+        return surveyKey;
+    }
+    public void setSurveyKey(String surveyKey) {
+        this.surveyKey = surveyKey;
+    }
```

Comment:
We need surveyKey in the table for look-ups, but use survey guid/createdOn to retrieve a survey, so these could be derived from the surveyKey. Will change this.